knowage-vue/src/modules/documentExecution/main/DocumentExecution.vue

```diff
@@ -1012,6 +1080,15 @@ export default defineComponent({
             })
 
             sessionStorage.setItem(this.document.label, JSON.stringify(tempFilters))
+        },
+        async onCrossNavigationSelected(event: any) {
+            this.destinationSelectDialogVisible = false
+            await this.loadCrossNavigation(event, this.angularData)
+        },
+        onCrossNavigationContainerClose() {
+            this.crossNavigationContainerData = null
+            this.crossNavigationContainerVisible = true
+            this.onBreadcrumbClick(this.breadcrumbs[0])
         }
     }
 })
```

Comment:
As a reminder, the complexity of over 1000 lines of code is excessive. For new developments or bug fix, please remember to always try to split code and no longer add it to this file. Thank you.

Reply:
Thanks, I will try, Document Execution got a lot bigger over time (OLAP was added, Cross Navigation changes etc.)